src/mixins/java/org/spongepowered/common/mixin/api/minecraft/world/level/LevelMixin_API.java

```diff
@@ -308,7 +308,23 @@ public Collection<? extends BlockEntity> blockEntities() {
 
     @Override
     public void addBlockEntity(final int x, final int y, final int z, final BlockEntity blockEntity) {
-        this.shadow$setBlockEntity(new BlockPos(x, y, z), (net.minecraft.world.level.block.entity.BlockEntity) Objects.requireNonNull(blockEntity, "blockEntity"));
+        // So, here we basically want to copy the given block entity into the location.
+        // BlockEntity stores its location, as well as it being mutable and stuff, so just setting what we've given here
+        // would cause unexpected bugs.
+        final net.minecraft.world.level.block.entity.BlockEntity mcOriginalBlockEntity = (net.minecraft.world.level.block.entity.BlockEntity) Objects.requireNonNull(blockEntity, "blockEntity");
+        // Save the nbt so we can copy it
+        final CompoundTag tag = mcOriginalBlockEntity.save(new CompoundTag());
+        // Ensure that where we are placing this blockentity is the right blockstate, so that minecraft will actually accept it.
+        this.world().setBlock(x, y, z, (org.spongepowered.api.block.BlockState) mcOriginalBlockEntity.getBlockState());
```

Comment:
Same with this, it's not a controlled set block, I'd like to have better control with using the block flags to set the block itself and ensure that lightning is enabled but drops are not enabled.